### ISSUE TITLE
Different output using hic and cool format

### ISSUE THREAD

**Issue by yhx-tj** [ay-lab/mustache#26]
Hi,
I'm using mustache (v1.2.0) to call HFFc6 Micro-C loops and I ran into the same issue mentioned in #20. I downloaded the updated hic and cool data from 4DN(4DNFIPC7P27B.hic & 4DNFI9FVHJZQ.mcool). I used the same options (-r 5kb -pt 0.01) but then it reported 28,007 and 36,554 loops from .mcool and .hic file, respectively. As mentioned in https://data.4dnucleome.org/resources/data-analysis/hi_c-processing-pipeline, the two files are generated from the same pairs file as input filtered contact list.

So which loop list should I choose?

Thank you so much for your help!

**Comment by ay-lab** [comment]
Hi, thanks for using our tool and pointing out this issue. Just to make sure, can you make sure that you are using the same normalization method for both files (maybe by explicitly stating -norm KR)? I will download the files and look into the problem.

**Comment by yhx-tj** [comment]
Yes, I uesd KR normalization for .hic files:

`mustache -f 4DNFI9FVHJZQ.mcool -o 4DNFI9FVHJZQ_output.tsv -r 5kb -pt 0.01
mustache -f 4DNFIPC7P27B.hic -o 4DNFIPC7P27B_output.tsv -norm KR -ch 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X -r 5kb -pt 0.01`

**Comment by ay-lab** [comment]
I got 36559 loops for 4DNFI9FVHJZQ.mcool. Just add the -norm KR for mcool format too. Also, for .hic data you don't need to list all the chromosomes. It will run automatically on all the chromosomes if you don't use -ch.

**Comment by yhx-tj** [comment]
Thank you so for your time! So the difference was caused by the ICE and KR normalization...I thought they were quite the same. And the description of the -ch option in the command line did not change so I missed the update, thanks again for pointing this out.

**Comment by yhx-tj** [comment]
Sorry, I got 51,036 loops for 4DNFI9FVHJZQ.mcool after adding -norm option:

`mustache -f 4DNFI9FVHJZQ.mcool -o 4DNFI9FVHJZQ_KR_output.tsv -norm KR -r 5kb -pt 0.01`

The log:
[mustache_call.log](https://github.com/ay-lab/mustache/files/7249234/mustache_call.log)

**Comment by roayaei** [comment]
The first time, I used the file I had downloaded (4DNFI9FVHJZQ.mcool) about two years ago so I thought maybe something has changed. So, I downloaded it again and ran mustache and got exactly the same (36,558) number of loops. Can you please make sure you are using the latest version of mustache and also that the file is the right file (mine is 11099585140 bytes).

**Comment by yhx-tj** [comment]
Yes, my mcool file is 11099585140 bytes too. And I just updated mustache to the latest version with pip two days ago. 

`cat ~/anaconda3/lib/python3.8/site-packages/mustache/_version.py`
`__version__ = "1.2.0"`

**Comment by ay-lab** [comment]
That's interesting. I ran mustache with and without -norm KR (with two different mustache versions) and none of them produced 51k. My only guess is that maybe you are still using an older version (maybe you are not activating the env that you have updated mustache in). Can you update your code using "git pull" and run mustache with "python mustache/mustache.py -f  4DNFI9FVHJZQ.mcool -norm KR -pt 0.01 -o test -r 5000 " instead.

**Comment by roayaei** [comment]
Hi, I am not sure about the reason behind the small difference (maybe the juicer data and cooler data are slightly different or maybe due to different precisions...). Yes, I think I was getting 28k for ICE.

**Comment by yhx-tj** [comment]
Thank you very much!
